### PR TITLE
Add inspect sub-command with basic support

### PR DESCRIPTION
**Pull request by doru1004**
**Release Note**
```release-note
kind/feature
```

This patch adds a new subcommand called `inspect`. This subcommand lists the dependencies for a given list of integration files. This patch adds basic support by printing out the list of top-level dependencies.

This patch partially addresses issue https://github.com/apache/camel-k/issues/1738

### CODE REVIEW

File: pkg/util/util.go [hunk 1]
```diff
@@ -219,6 +219,15 @@ func FileExists(name string) (bool, error) {
 	return !info.IsDir(), err
 }
 
+// DirectoryExists --
+func DirectoryExists(directory string) (bool, error) {
+	info, err := os.Stat(directory)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return info.IsDir(), err
```
Comment (lburgazzoli) [inline pkg/util/util.go:228]: if err != nil, then info could be nil so maybe better to check this case

File: pkg/cmd/inspect.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+				if util.StringSliceExists(v.Languages, string(lang)) {
+					dependencies.Add(fmt.Sprintf("mvn:%s/%s", v.GroupID, v.ArtifactID))
+
+					for _, d := range v.Dependencies {
+						dependencies.Add(fmt.Sprintf("mvn:%s/%s", d.GroupID, d.ArtifactID))
+					}
+				}
+			}
+		}
+	}
```
Comment (lburgazzoli) [inline pkg/cmd/inspect.go:177]: wonder if this block could be extricated in an utility method so so it won't break in case the logic changes

**Review by lburgazzoli — CHANGES_REQUESTED** [review]

**Comment by doru1004** [comment]
@lburgazzoli I think I addressed all the comments. Please let me know if further changes are required.

File: pkg/cmd/inspect.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+func generateCatalog() (*camel.RuntimeCatalog, error) {
+	// A Camel catalog is requiref for this operatio.
+	settings := ""
+	mvn := v1.MavenSpec{
+		LocalRepository: "",
+	}
+	runtime := v1.RuntimeSpec{
+		Version:  "1.3.0",
```
Comment (lburgazzoli) [inline pkg/cmd/inspect.go:177]: Guess this version can be taken from: https://github.com/apache/camel-k/blob/master/pkg/util/defaults/defaults.go#L29

File: pkg/cmd/inspect.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+	return &cmd, &options
+}
+
+type inspectCmdOptions struct {
+	*RootCmdOptions
+	AllDependencies        bool   `mapstructure:"all-dependencies"`
+	OutputFormat           string `mapstructure:"output"`
+	DependenciesDirectory  string `mapstructure:"dependencies-directory"`
+	AdditionalDependencies string `mapstructure:"additional-dependencies"`
```
Comment (lburgazzoli) [inline pkg/cmd/inspect.go:70]: looks like `AllDependencies`, `DependenciesDirectory`, `AdditionalDependencies` are not taken into account at this stage, can we remove them ?

**Comment by doru1004** [comment]
@lburgazzoli just resolved all additional comments.

**Review by lburgazzoli — APPROVED** [review]

**Comment by nicolaferraro** [comment]
> @lburgazzoli just resolved all additional comments.

Thanks @doru1004 ! Can you rebase on master so we can merge this? there are some conflicts.

**Comment by doru1004** [comment]
Absolutely, doing this right now. :)

**Comment by doru1004** [comment]
That was the last of the changes, resources.go should be correctly updated now.

**Review by nicolaferraro — APPROVED** [review]

**Pull request merged**